### DISCUSSION TITLE
Would you like type hints for this project?

### DISCUSSION THREAD

**Discussion by demberto** [Zulko/moviepy#1960] [Ideas]
I see a few open issues for this (#1695, #1912), but apart from that there seems to be no work on it. Would you like a PR for type hints?
I could also submit type stubs to `typeshed` if maintainers aren't yet ready to factor in the changes introduced to the codebase. Let me know.

**Comment by mgaitan** [comment]
@demberto it would be cool, yeah. Also implementing mypy as part of CI.

**Reply by demberto** [reply]
Would you like them as a part of the codebase (aka inlined) or separated (in the form of stubs)?

**Comment by and-oli** [comment]
maybe a long shot given the age of this discussion but having type annotations would be great! I think it would be more convenient if types are part of the code itself but stubs would still be super useful!
